src/schema/term.rs

```diff
@@ -42,18 +30,50 @@ impl AsMut<Vec<u8>> for Term {
     }
 }
 
+/// The number of bytes used as metadata by `Term`.
+const TERM_METADATA_LENGTH: usize = 5;
+
 impl Term {
-    pub(crate) fn new() -> Term {
-        Term(Vec::with_capacity(100))
+    pub(crate) fn with_capacity(capacity: usize) -> Term {
+        let mut data = Vec::with_capacity(TERM_METADATA_LENGTH + capacity);
+        data.resize(TERM_METADATA_LENGTH, 0u8);
+        Term(data)
+    }
+
+    pub(crate) fn with_type_and_field(typ: Type, field: Field) -> Term {
+        let mut term = Self::with_capacity(8);
```

Comment:
It's a magic number, but it doesn't affect correctness. It's on the lower end on how much payload we expect.

We can `resize` and set or `extend`, but `reserve` and will not change the vec's length.

Reply:
> We can `resize` and set or `extend`, but `reserve` and will not change the vec's length.

Indeed the suggestion is incorrect. My question would then be whether a special case that pushes the field and type instead of setting would be reasonable based on how often this constructor is called?

Reply:
I don't know if that makes a relevant performance difference